Client/Frontend/Browser/TabPeekViewController.swift

```diff
@@ -0,0 +1,132 @@
+                guard let displayURL = self.tab.url?.absoluteString where displayURL.characters.count > 0 else { return }
+                self.bvc?.addToReadingList(displayURL, title: self.tab.lastTitle ?? displayURL)
+            })
+        }
+        if !self.isBookmarked {
+            actions.append(UIPreviewAction(title: self.PreviewActionAddToBookmarks, style: .Default) { previewAction, viewController in
+                guard let displayURL = self.tab.url?.absoluteString where displayURL.characters.count > 0 else { return }
+
+                self.bvc?.addBookmark(displayURL, title: self.tab.lastTitle ?? displayURL)
+                })
+        }
+        if self.hasRemoteClients {
+            actions.append(UIPreviewAction(title: self.PreviewActionSendToDevice, style: .Default) { previewAction, viewController in
+                guard let clientPicker = self.clientPicker else { return }
+                self.bvc?.presentViewController(clientPicker, animated: false, completion: nil)
+                })
+        }
+        actions.append(UIPreviewAction(title: self.PreviewActionCopyURL, style: .Default) { previewAction, viewController in
+            guard let url = self.tab.url where url.absoluteString.characters.count > 0 else { return }
+            let pasteBoard = UIPasteboard.generalPasteboard()
+            pasteBoard.URL = url
+            })
+        actions.append(UIPreviewAction(title: self.PreviewActionCloseTab, style: .Destructive) { previewAction, viewController in
+            guard let tabViewController = viewController as? TabPeekViewController else { return }
+            self.tabManager.removeTab(self.tab)
+            })
+
+        return actions
+    }()
+
+
+    init(tab: Browser, controller: BrowserViewController?, tabManager: TabManager) {
+        self.tab = tab
+        self.bvc = controller
+        self.tabManager = tabManager
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // if there is no screenshot, load the URL in a web page
+        // otherwise just show the screenshot
+        guard let screenshot = tab.screenshot else {
+            setupWebView(tab.url)
+            return
+        }
+        setupWithScreenshot(screenshot)
+    }
+
+    private func setupWithScreenshot(screenshot: UIImage) {
+        let imageView = UIImageView(image: screenshot)
+        self.view.addSubview(imageView)
+
+        imageView.snp_makeConstraints { make in
+            make.edges.equalTo(self.view)
+        }
+    }
+
+    private func setupWebView(url: NSURL?) {
+        let webView = WKWebView()
+        self.view.addSubview(webView)
+
+        webView.snp_makeConstraints { make in
+            make.edges.equalTo(self.view)
+        }
+
+        if let url = url {
+            webView.loadRequest(NSURLRequest(URL: url))
```

Comment:
Note that using this on a private browsing tab will cause that tab to load here without the private process pool, meaning cookies and other data will be leaked. We should either reuse the tab's configuration here or not follow the `setupWebView` path for private tabs.